File: CR12/Lectures13-14/approxTensor.m
function T = approxTensor(L, U1, U2, U3)

	ncols  = length(L);
	
	n1 = rows(U1);
	n2 = rows(U2);
	n3 = rows(U3);
	
	T = zeros(n1, n2, n3);

	
	
	for i=1:ncols
		T += L(i) * outerproduct(U1(:,i), U2(:,i), U3(:,i));
	end

end

File: CR12-2024/Lecture14/approxtensor.py
from outerproduct import *

def approxtensor(L, U1, U2, U3):
    ncols = len(L)

    n1 = U1.shape[0]
    n2 = U2.shape[0]
    n3 = U3.shape[0]

    T = np.zeros((n1, n2, n3))

    for k in range(ncols):
        T += L[k] * outerproduct(U1[:,k], U2[:, k], U3[:, k])

    return T
